scripts/gis-material-update/automation/automation.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/sh
+
+for tormays_source in $*
```

Comment:
Running both `hsl` and `tram_lines` on the same run causes the script to download the HSL data twice?

Reply:
Yes it does. There is no easy way to check in this level (bash script) that is the data already downloaded or not.

Reply:
You could have the check in `fetch_data.sh`. Maybe with an option to override the file anyway.

Another approach would be to run all the fetch phases at once, then all the process and deploy phases. Maybe something like (untested):
```shell
# Collect the sources needed for all targets:
sources=""
for tormays_source in $*
do
    case $tormays_source in
    hsl)
        sources="${sources} hsl hki"
        ;;
    tram_infra)
        sources="${sources} hki osm helsinki_osm_lines"
        ;;
    tram_lines)
        sources="${sources} hki hsl"
        ;;
    liikennevaylat)
        sources="${sources} liikennevaylat central_business_area ylre_katuosat"
        ;;
    *)
        sources="${sources} $tormays_source"
        ;;
    esac
done
# Deduplicate sources
sources=$echo "$sources" | xargs -n1 | sort -u | xargs)
# Fetch the sources
sh /haitaton-gis/fetch_all.sh ${sources}
```

Reply:
sources=$(echo "$sources" | xargs -n1 | sort -u | xargs)

Reply:
Changed